File: android/BertSpeak/app/src/main/java/chuckcoughlin/bert/MainActivityPagerAdapter.java
/**
 * Copyright 2019 Charles Coughlin. All rights reserved.
 *  (MIT License)
 */

package chuckcoughlin.bert;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.util.Log;

import chuckcoughlin.bert.tab.AssistantFragment;
import chuckcoughlin.bert.tab.CoverFragment;
import chuckcoughlin.bert.tab.DispatcherLogsFragment;
import chuckcoughlin.bert.tab.SettingsFragment;
import chuckcoughlin.bert.tab.TablesTabFragment;
import chuckcoughlin.bert.tab.TranscriptFragment;


/**
 * There is a specialized page fragment for each tab position.
 * Return the appropriate fragment when requested.
 */

public class MainActivityPagerAdapter extends FragmentStatePagerAdapter {
    private final static String CLSS = "MainActivityPagerAdapter";
    private String[] tabTitles;

    public MainActivityPagerAdapter(FragmentManager fm,Context ctx) {
        super(fm);

        tabTitles = new String[] {
                ctx.getString(R.string.cover_tab_label),ctx.getString(R.string.dispatcher_log_tab_label),
                ctx.getString(R.string.transcript_tab_label),ctx.getString(R.string.tables_tab_label),
                ctx.getString(R.string.settings_tab_label)};
        Log.i(CLSS,"Constructor ...");
    }

    /**
     * Each page is a different class.
     * @param position page number
     * @return a new instance of the page.
     */
    @Override
    public Fragment getItem(int position) {
        AssistantFragment frag = null;

        switch (position) {
            case 0:
                frag = new CoverFragment();
                break;
            case 1:
                frag = new DispatcherLogsFragment();
                break;
            case 2:
                frag = new TranscriptFragment();
                break;
            case 3:
                frag =  new TablesTabFragment();
                break;
            case 4:
                frag = new SettingsFragment();
                break;
            default:
        }
        if( frag!=null ) {
            Log.i(CLSS,"getItem: "+position+": fragment="+frag.getClass().getCanonicalName());
            frag.setPageNumber(position);
            frag.setTitle(tabTitles[position]);
        }
        return (Fragment)frag;
    }



    /**
     * @return the number of pages in our repertoire.
     */
    @Override
    public int getCount() {
        return tabTitles.length;
    }


    @Override
    public CharSequence getPageTitle(int position) {
        return tabTitles[position];
    }
}

File: BertSpeak/app/src/main/kotlin/chuckcoughlin/bertspeak/MainActivityPagerAdapter.kt
/**
 * Copyright 2022-2023 Charles Coughlin. All rights reserved.
 * (MIT License)
 */
package chuckcoughlin.bertspeak


import android.util.Log
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import chuckcoughlin.bertspeak.tab.AssistantFragment
import chuckcoughlin.bertspeak.tab.CoverFragment
import chuckcoughlin.bertspeak.tab.RobotLogsFragment
import chuckcoughlin.bertspeak.tab.SettingsFragment
import chuckcoughlin.bertspeak.tab.TablesTabFragment
import chuckcoughlin.bertspeak.tab.TranscriptFragment

/**
 * This is a specialized page fragment for each tab position.
 * Return the appropriate fragment when requested.
 */
class MainActivityPagerAdapter(fragManager: FragmentManager, lifecycle: Lifecycle,titles: Array<String>) : FragmentStateAdapter(fragManager,lifecycle) {
    private val tabTitles = titles
    /**
     * Each page is a different class. If position is out of range, use 0
     * @param position page number
     * @return a new instance of the page.
     */
    override fun createFragment(position: Int): Fragment {
        var pos = position
        var frag: AssistantFragment
        if (position<0) pos = 0
        if(position>tabTitles.size - 1) pos = tabTitles.size - 1
        when (pos) {
            0 -> frag = CoverFragment(pos)
            1 -> frag = TranscriptFragment(pos)
            2 -> frag = RobotLogsFragment(pos)
            3 -> frag = TablesTabFragment(pos)
            4 -> frag = SettingsFragment(pos)
            else ->
                frag = CoverFragment(pos)
        }

        Log.i(CLSS, "createFragment: " + position + ": fragment=" + frag.javaClass.canonicalName)
        frag.title = getPageTitle(position)
        return frag as Fragment
    }

    /**
     * @return the number of pages in our repertoire.
     */
    override fun getItemCount(): Int {
        return tabTitles.size
    }
    fun getPageTitle(position: Int): String {
        return tabTitles[position]
    }


    companion object {
        private const val CLSS = "MainActivityPagerAdapter"
    }
}
